src/basic/rm-rf.h

```diff
@@ -32,10 +32,10 @@ int rm_rf_children(int fd, RemoveFlags flags, struct stat *root_dev);
 int rm_rf(const char *path, RemoveFlags flags);
 
 /* Useful for usage with _cleanup_(), destroys a directory and frees the pointer */
-static inline void rm_rf_and_free(char *p) {
+static inline void rm_rf_physical_and_free(char *p) {
         if (!p)
```

Comment:
actually, even though the new name is a slight misnomer, I think i prefer if it makes clear that this does include removal of physical. i.e. it should mention this somewhere, and I think "rm_rf_physical_and_free" is good enough...

I mean, we could also try to be fully correct, but then we should probably call it rm_rf_with_root_and_physical_and_free(), but that'll be too complex I think. Hence I think we should just merge your patch as-is.